hvplot/__init__.py

```diff
@@ -55,21 +55,36 @@ def _get_doc(cls, kind, completions=False, docstring=True, generic=True, style=T
         defaults = argspec.defaults or [None]*len(args)
         for arg, dflt in zip(args, defaults):
             parameters.append((arg, dflt))
+        signature = None
     else:
         sig = inspect.signature(method)
         for name, p in list(sig.parameters.items())[1:]:
             if p.kind == 1:
                 parameters.append((name, p.default))
+
+        extra_kwargs = _hv.core.util.unique_iterator(valid_opts + kind_opts
+                                                     + converter._axis_options
+                                                     + converter._op_options)
+        filtered_signature = [p for p in sig.parameters.values()
+                              if p.kind != inspect.Parameter.VAR_KEYWORD]
+        extra_params = [inspect.Parameter(k, inspect.Parameter.KEYWORD_ONLY)
+                        for k in extra_kwargs
+                        if k not in [p.name for p in filtered_signature]]
+        all_params = (filtered_signature + extra_params
+                      + [inspect.Parameter('kwargs', inspect.Parameter.VAR_KEYWORD)])
+        signature = inspect.Signature(all_params)
+
     parameters += [(o, None) for o in
                    valid_opts+kind_opts+converter._axis_options+converter._op_options]
```

Comment:
Should the extra_kwargs processing and this not be merged somehow? Seems like they would be duplicated here as well.

Reply:
I assume you are talking about `parameters` in which case yes...there will be duplication in the docstring completion hint. The difference is that duplication in the docstring hint doesn't generate an error (and tab completion works fine) while the signature approach validates against duplication.

I can use `unique_iterator` here but I'm not sure it is worth doing anything much more complicated: we will be dropping Python 2 before long (so we won't be using the docstring hint anymore) and the only improvement would be a slight reduction in the verbosity of part of the docstring no one looks at anyway...

Reply:
Addressed in bbf6f61.